scipio/src/io/mod.rs

```diff
@@ -86,17 +86,14 @@ use std::path::Path;
 /// rename an existing file.
 ///
 /// Warning: synchronous operation, will block the reactor
-pub async fn rename_file<P: AsRef<Path>>(old_path: P, new_path: P) -> io::Result<()> {
-    let new_path = new_path.as_ref().to_owned();
-    let old_path = old_path.as_ref().to_owned();
-
-    sys::rename_file(&old_path, &new_path)
+pub async fn rename<P: AsRef<Path>>(old_path: P, new_path: P) -> io::Result<()> {
```

Comment:
For full generality, this should have two type parameters: https://doc.rust-lang.org/stable/std/fs/fn.rename.html

Otherwise, LGTM.

Reply:
I find this a bit weird but I am all for conformity.

Thanks for bringing that to my attention

Reply:
With one type parameter, two types must match exactly, for expelled, &str and &str or &Path and &Path. With two type parameters, you can mix different types, like &str and &Path.

So this is rather API flexibility than conformity.